api/lib/domain/events/handle-badge-acquisition.js

```diff
@@ -7,37 +7,36 @@ const handleBadgeAcquisition = async function({
   campaignParticipationResultRepository,
 }) {
   if (completedAssessmentBelongsToACampaign(assessmentCompletedEvent)) {
-    const badge = await fetchPossibleCampaignAssociatedBadge(assessmentCompletedEvent, badgeRepository);
-    if (isABadgeAssociatedToCampaign(badge)) {
-      const campaignParticipationResult = await fetchCampaignParticipationResults(assessmentCompletedEvent, badge, campaignParticipationResultRepository);
-      if (isBadgeAcquired(campaignParticipationResult, badge.badgeCriteria, badgeCriteriaService)) {
-        await badgeAcquisitionRepository.create({
+    const campaignBadges = await fetchPossibleCampaignAssociatedBadges(assessmentCompletedEvent, badgeRepository);
+    const campaignParticipationResult = await fetchCampaignParticipationResults(assessmentCompletedEvent, campaignBadges, campaignParticipationResultRepository);
+
+    const badgeAquisitionsToCreate = campaignBadges
+      .filter((badge) => isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService))
+      .map((badge) => {
+        return badgeAcquisitionRepository.create({
           badgeId: badge.id,
           userId: assessmentCompletedEvent.userId
         }, domainTransaction);
```

Comment:
Même remarque que https://github.com/1024pix/pix/pull/1475/files#r434386080 pour le `Promise.all`. D'ailleurs `badgeAquisitionsToCreate` devrait s'appeler `badgeAcquisitionsBeingCreated` pour mieux traduire le fait qu'on a lancé _n_ requêtes en parallèle (👎) et que le `Promise.all` ne fait qu'attendre la fin de leur exécution.

Donc, version avec un utilitaire `bluebird` :

```suggestion
    const acquiredBadges = campaignBadges
      .filter((badge) => isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService));

    await bluebird.each(acquiredBadges, (badge) => {
      return badgeAcquisitionRepository.create({
        badgeId: badge.id,
        userId: assessmentCompletedEvent.userId
      }, domainTransaction);
    });
```

Version avec `for`, qui est bien aussi (surtout quand on n'a pas de valeur de retour, comme ici) :

```suggestion
	for (badge of acquiredBadges) {
	  if (isBadgeAcquired(campaignParticipationResult, badge, badgeCriteriaService)) {
        await badgeAcquisitionRepository.create({
          badgeId: badge.id,
          userId: assessmentCompletedEvent.userId
        }, domainTransaction);
	  }
	}
```

Reply:
La méthode `badgeAcquisitionRepository.create` a été modifiée pour créer plusieurs badges-acquisitions à la fois. Ca évite d'avoir recours à une boucle asynchrone dans `handle-badge-acquisition.js`